engine/crates/integration-tests/src/postgresql.rs

```diff
@@ -0,0 +1,207 @@
+            .registry
+            .export_sdl(false);
+
+        parse_schema::<String>(&result).unwrap().to_string()
+    })
+}
+
+#[track_caller]
+fn inner_query_neon<P, L, U, R>(api: U, database: &str, test: P) -> String
+where
+    P: FnOnce(TestApi) -> L,
+    L: Future<Output = Response>,
+    U: FnOnce() -> R,
+    R: Future<Output = TestApi>,
+{
+    super::runtime().block_on(async {
+        let admin = NeonTransport::new(ADMIN_CONNECTION_STRING).unwrap();
+
+        admin
+            .execute(&format!("DROP DATABASE IF EXISTS {database}"))
+            .await
+            .unwrap();
+
+        admin.execute(&format!("CREATE DATABASE {database}")).await.unwrap();
+
+        let api = api().await;
+        let response = AssertUnwindSafe(test(api.clone())).catch_unwind().await;
+
+        let response = response.expect("Error in test execution.");
+
+        serde_json::to_string_pretty(&response.to_graphql_response()).unwrap()
+    })
+}
+
+pub struct Inner {
+    engine: OnceCell<Engine>,
+    connection: NeonTransport,
+    schema: String,
+}
+
+#[derive(Clone)]
+pub struct TestApi {
+    inner: Arc<Inner>,
+}
+
+impl TestApi {
+    async fn new(database: &str) -> Self {
+        let mut url = url::Url::parse(ADMIN_CONNECTION_STRING).unwrap();
+        url.set_path(&format!("/{database}"));
+
+        let connection_string = url.to_string();
+
+        let schema = formatdoc! {r#"
+            extend schema
+              @neon(
+                name: "test",
+                url: "{connection_string}",
+                namespace: false
+              )    
+        "#};
+
+        Self::new_inner(schema, connection_string).await
+    }
+
+    async fn new_namespaced(database: &str, name: &str) -> Self {
+        let mut url = url::Url::parse(ADMIN_CONNECTION_STRING).unwrap();
+        url.set_path(&format!("/{database}"));
+
+        let connection_string = url.to_string();
+
+        let schema = formatdoc! {r#"
+            extend schema
+              @neon(
+                name: "{name}",
+                url: "{connection_string}",
+                namespace: true
+              )
+        "#};
+
+        Self::new_inner(schema, connection_string).await
+    }
+
+    async fn new_inner(schema: String, connection_string: String) -> Self {
+        let engine = OnceCell::new();
+        let connection = NeonTransport::new(&connection_string).unwrap();
+
+        let inner = Inner {
+            engine,
+            connection,
+            schema,
+        };
+
+        Self { inner: Arc::new(inner) }
+    }
+
+    pub async fn execute_sql(&self, query: &str) -> ExecuteResponse {
+        self.inner
+            .connection
+            .execute(query)
+            .await
+            .expect("error in query execute")
+    }
+
+    pub async fn execute(&self, operation: impl AsRef<str>) -> Response {
+        self.inner
+            .engine
+            .get_or_init(async { Engine::new(self.inner.schema.clone()).await })
```

Comment:
This prevents a race. If we initialize the `Engine` together with `TestApi`, it runs the introspection right away, and after that gives the control to the test that can create tables and so on. The engine doesn't know anything about these tables, so we postpone the initialization until the first GraphQL query/mutation arrives.

Reply:
worth mentioning in a comment then if you bothered making a comment in GitHub.